tests/k8st/tests/test_bgp_advert.py

```diff
@@ -354,21 +366,34 @@ def test_rr(self):
 """)
         svc_json = kubectl("get svc nginx-rr -n bgp-test -o json")
         svc_dict = json.loads(svc_json)
-        svcRoute = svc_dict['spec']['clusterIP']
-        retry_until_success(lambda: self.assertIn(svcRoute, self.get_routes()))
+        cluster_ip = svc_dict['spec']['clusterIP']
+        external_ip = svc_dict['spec']['externalIPs'][0]
+        retry_until_success(lambda: self.assertIn(cluster_ip, self.get_routes()))
+        retry_until_success(lambda: self.assertIn(external_ip, self.get_routes()))
 
-    def test_mainline(self):
+    def test_cluster_ip_advertisement(self):
         """
-        Runs the mainline tests for service ip advertisement
+        Runs the tests for service cluster IP advertisement
         - Create both a Local and a Cluster type NodePort service with a single replica.
           - assert only local and cluster CIDR routes are advertised.
           - assert /32 routes are used, source IP is preserved.
-        - Create a local LoadBalancer service with clusterIP = None, assert no change.
         - Scale the Local NP service so it is running on multiple nodes, assert ECMP routing, source IP is preserved.
         - Delete both services, assert only cluster CIDR route is advertised.
         """
         with DiagsCollector():
-            # Assert that a route to the service IP range is present.
+
+            calicoctl("""apply -f - << EOF
```

Comment:
Do we need to delete this BGPConfiguration after the test? (Or maybe we already have something that does that?)

Reply:
The bgpconfig is reset in tearDown